### PR TITLE
[Reset Password v1] Refactor ForcePasswordReset into AuthResult

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes
- **login.component**: Added conditional/options for navigation when forcing a password reset is necessary
- **sso.component**:  Added conditional/options for navigation when forcing a password reset is necessary
- **two-factor.component**: Added conditional for when forcing a password reset is necessary
- **authResult**: Added `forcePasswordReset` boolean
- **identityTokenResponse**: Added `forcePasswordReset` boolean
- **auth.service**: Added `forcePasswordReset` to the `AuthResult` response object

### CODE REVIEW

File: angular/src/components/sso.component.ts [hunk 1]
```diff
@@ -161,6 +163,12 @@ export class SsoComponent {
                         },
                     });
                 }
+            } else if (response.forcePasswordReset) {
```
Comment (Hinton) [inline angular/src/components/sso.component.ts:166]: We are starting to have very many branches in this function. Which makes it a bit tricky to follow.

One thing that comes to mind that would clean it up a fair bit would be to convert the `on<hook>` to actual methods with a default implementation.

I.e. 

```ts
protected onSuccessfulLoginForceResetNavigate(orgIdFromState) {
    this.router.navigate([this.changePasswordRoute], {
        queryParams: {
            identifier: orgIdFromState,
        },
    });
}
```



 We could simplify things by using proper functions for the `onSuccessfulLoginForceResetNavigate` which by default does `this.router.navigate([this.forcePasswordResetRoute]);`.

We could then look into using guard clauses.

```ts
this.loggingIn = true;
try {
    this.formPromise = this.authService.logInSso(code, codeVerifier, this.redirectUri);
    const response = await this.formPromise;
    
    if (response.twoFactor) {
        this.onSuccessfulLoginTwoFactorNavigate();
        return;
    }

    ...
}
```

**Review by Hinton — APPROVED** [review]
In general it looks good, I wonder if we should take the opportunity to clean up the `sso.component` a bit, to avoid having nested if cases.

File: angular/src/components/sso.component.ts (hunk 1 above)
Comment (vincentsalucci) [reply angular/src/components/sso.component.ts:166]: Noted - will tackle some of this cleanup in another group of PRs.

**Comment by vincentsalucci** [comment]
@Hinton - Sorry for the dismissal, the branch was getting a little stale (my fault).

**Review by Hinton — APPROVED** [review]

File: angular/src/components/sso.component.ts (hunk 1 above)
Comment (Hinton) [reply angular/src/components/sso.component.ts:166]: Feel free to make a tech debt item and throw it on Asana if you don't have time atm.

**Pull request merged**